### PR TITLE
Remove gnustl support and update NDK

### PR DESCRIPTION
**Pull request by jonsimantov**
For next major revision, remove gnustl support and update to NDK r21e.

In the SDK binary distribution, Android libraries are now in simply libs/android/ARCH, with no STL path component.
***

Running [integration tests](https://github.com/firebase/firebase-cpp-sdk/pull/889#issuecomment-1095348828) in the PR.
Running [c++ packaging workflow](https://github.com/firebase/firebase-cpp-sdk/actions/workflows/cpp-packaging.yml?query=branch%3Adeprecation%2Fremove-gnustl) on the branch and confirming Android libraries are placed in the correct location.
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [X] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by jonsimantov** [comment]
(Note: Will not check this in until we are preparing for the major revision bump.)

File: release_build_files/readme.md [hunk 1]
```diff
@@ -53,9 +53,8 @@ libs/ios/ARCH             | iOS static libraries (compiled against
 libs/android/ARCH/STL     | Android (GCC 4.8+ compatible) static
 |                         | libraries for each architecture and STL
 |                         | variant.
-|                         | _STL variants available:_
+|                         | _STL variant available:_
 |                         | * `c++`: LLVM libc++ runtime (recommended)
```
Comment (dconeybe) [inline release_build_files/readme.md:57]: Drive-by comment: Remove "(recommended)" from this line, since it's the _only_ option now.

File: build_scripts/android/build.sh [hunk 2]
```diff
@@ -16,11 +16,11 @@ if [[ ! -d "${sourcepath}" ]]; then
     exit 2
 fi
 
-if [[ "${stl}" == "c++" || "${stl}" == "gnustl" ]]; then
+if [[ "${stl}" == "c++" ]]; then
```
Comment (a-maurice) [inline build_scripts/android/build.sh:19]: Do we want to just get rid of the ability to specify stl, since there is only one option anyway?

Comment (jonsimantov) [reply build_scripts/android/build.sh]: Sure, let's give it a try. I also decided to remove STL from the Android library path, will run packaging to make sure everything ends up in the right place.

File: build_scripts/android/build.sh [hunk 3]
```diff
@@ -4,10 +4,9 @@
 
 buildpath=$1
 sourcepath=$2
-stl=$3
 
 if [[ -z "${buildpath}" || -z "${sourcepath}" ]]; then
-    echo "Usage: $0 <build path> <source path> [c++|gnustl]"
+    echo "Usage: $0 <build path> <source path> [c++]"
```
Comment (a-maurice) [inline build_scripts/android/build.sh:9]: We can drop the c++ bit now

**Review by a-maurice — APPROVED** [review]

File: build_scripts/android/build.sh (hunk 3 above)
Comment (jonsimantov) [reply build_scripts/android/build.sh:9]: Will update at some other point, since this auto-merged.
